src/main/MQ2ImGuiConsole.cpp

```diff
@@ -1328,6 +1348,49 @@ static void MakeColorGradient(float frequency1, float frequency2, float frequenc
 }
 
 //----------------------------------------------------------------------------
+ImGuiWindowFlags ConsoleFlagSetup()
+{
+	ImGuiWindowFlags windowFlags = 0;
+	if (!s_noTitleBar || s_mouseHoveringConsoleTimer != 0) windowFlags |= ImGuiWindowFlags_MenuBar;
+	if (s_noTitleBar)  windowFlags |= ImGuiWindowFlags_NoTitleBar;
+	if (s_lockConsole) windowFlags |= ImGuiWindowFlags_NoMove | ImGuiWindowFlags_NoResize;
+	return windowFlags;
+}
```

Comment:
This function should be member function of the console. Its name should be more indicative of its behavior, so something like `GetConsoleFlags` would be more appropriate.